pkg/loader/load_test.go

```diff
@@ -378,6 +378,141 @@ func (s *batchManagerSuite) TestShouldExecAccumulatedDMLs(c *check.C) {
 	c.Assert(bm.txns, check.HasLen, 1)
 }
 
+type txnManagerSuite struct{}
+
+var _ = check.Suite(&txnManagerSuite{})
+
+func (s *txnManagerSuite) TestRunTxnManager(c *check.C) {
+	input := make(chan *Txn)
+	dmls := []*DML{
+		{Tp: UpdateDMLType},
+		{Tp: InsertDMLType},
+		{Tp: UpdateDMLType},
+	}
+	txn := &Txn{DMLs: dmls}
+	txnManager := newTxnManager(14, input)
+	output := txnManager.run()
+	// send 5 txns (size 3) to txnManager, the 5th txn should get blocked at cond.Wait()
+	for i := 0; i < 5; i++ {
+		select {
+		case input <- txn:
+		case <-time.After(50 * time.Microsecond):
+			c.Fatal("txnManager gets blocked while receiving txns")
```

Comment:
If the 5th txn get blocked sending to `input`, why didn't this test fail?

Reply:
The 5th txn was picked from `input` but can't be added to the `txnManager.cacheChan`. Although it gets blocked at cond.Wait() but it's still picked out from `input`. By the way, we can't know a `txn`'s size when it is not picked out from `input`.

Reply:
The check `c.Assert(output, check.HasLen, 4)` can make sure it's not added.